src/main/java/ch/njol/skript/expressions/ExprElement.java

```diff
@@ -60,28 +66,38 @@ public class ExprElement extends SimpleExpression<Object> {
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		expr = LiteralUtils.defendExpression(exprs[1]);
 		number = (Expression<Number>) exprs[0];
-		element = parseResult.mark;
+		type = ElementType.values()[parseResult.mark];
 		return LiteralUtils.canInitSafely(expr);
 	}
 
 	@Override
 	@Nullable
 	protected Object[] get(Event e) {
-		Object[] os = expr.getArray(e);
-		if (os.length == 0)
+		Iterator<?> iter = expr.iterator(e);
+		if (iter == null || !iter.hasNext())
 			return null;
-		Object o;
-		if (element == -1) {
-			o = os[0];
-		} else if (element == 1) {
-			o = os[os.length - 1];
-		} else if (element == 2) {
-			Number number = this.number.getSingle(e);
-			if (number == null || number.intValue() - 1 >= os.length || number.intValue() - 1 < 0)
-				return null;
-			o = os[number.intValue() - 1];
-		} else {
-			o = CollectionUtils.getRandom(os);
+		Object o = null;
+		switch (type) {
+			case FIRST:
+				o = iter.next();
+				break;
+			case LAST:
+				o = Iterators.getLast(iter);
+				break;
+			case ORDINAL:
+				assert this.number != null;
+				Number number = this.number.getSingle(e);
+				if (number == null)
+					return null;
+				try {
+					o = Iterators.get(iter, number.intValue() - 1);
+				} catch (IndexOutOfBoundsException exception) {
```

Comment:
_Just some idea came onto my mind_ - do you think it would be nice to allow negative numbers to return from last elements instead? just like python

Reply:
it's a nice idea, but i think 75% of the value of that is easily grabbing the last element which this expression already lets you do

Reply:
Probably. It's just easier to do something like `the -3rd element of {objs::*}` to get the 3rd element from the end of the list rather than `the (size of {objs::*} - 3)rd element of {objs::*}`

Reply:
I don't think it'd fit Skript well, it'd be better to create a syntax (probably another pattern for ExprElement), e.g. `2nd to last element of {_l::*}`

Reply:
added